### PR TITLE
Freshness pass on C++ docs

### PR DESCRIPTION
**Pull request by AlexandraKemperMS**
Updating the C++ docs, for now the MinGW on Windows doc, might add more to this PR or do them as a separate PR.

### CODE REVIEW

File: docs/cpp/config-mingw.md [hunk 1]
```diff
@@ -138,7 +159,11 @@ You'll only be asked to choose a compiler the first time you run `helloworld.cpp
 
     ![screenshot of program output](images/playbutton/helloworld-terminal-output.png)
 
-The first time you run your program, the C++ extension creates `tasks.json`, which you'll find in your project's `.vscode` folder. `tasks.json` stores build configurations.
+Congradulations! You've just run your first C++ program in VS Code!
```
Comment (roblourens) [inline docs/cpp/config-mingw.md:162]: "congratulations"

**Comment by isidorn** [comment]
@AlexandraKemperMS there are some conflicts. Once you resolve those and tackle Rob's comments we can merge this in.

File: docs/cpp/config-mingw.md [hunk 2]
```diff
@@ -3,58 +3,76 @@ Order: 3
...
+1. Install the [C/C++ extension for VS Code](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools). You can install the C/C++ extension by searching for 'C++' in the Extensions view (`kb(workbench.view.extensions)`).
 
     ![C/C++ extension](images/cpp/cpp-extension.png)
 
-1. Get the latest version of Mingw-w64 via [MSYS2](https://www.msys2.org/), which provides up-to-date native builds of GCC, Mingw-w64, and other helpful C++ tools and libraries. You can download the latest installer from the MSYS2 page or use this [link to the installer](https://github.com/msys2/msys2-installer/releases/download/2022-06-03/msys2-x86_64-20220603.exe).
+## Installing the MinGW-w64 toolchain
+Get the latest version of MinGW-w64 via [MSYS2](https://www.msys2.org/), which provides up-to-date native builds of GCC, MinGW-w64, and other helpful C++ tools and libraries. This will provide you with the necessary tools to compile your code, debug it, and configure it to work with [IntelliSense](/editor/intellisense).
```
Comment (gregvanl) [inline docs/cpp/config-mingw.md:29]: Use relative path from vscode-docs root and .md filename extension for better navigation for readers on GitHub.com
/docs/editor/intellisense.md

File: docs/cpp/config-mingw.md [hunk 3]
```diff
@@ -105,23 +123,26 @@ Now press `kb(workbench.action.files.save)` to save the file. Notice how the fil
...
+You can also enable [Auto Save](/docs/editor/codebasics.md#save-auto-save) to automatically save your file changes, by selecting **File** > **Auto Save**.  You can find out more about the other views in the VS Code [User Interface documentation](/docs/getstarted/userinterface.md).
 
 >**Note**: When you save or open a C++ file, you may see a notification from the C/C++ extension about the availability of an Insiders version, which lets you test new features and fixes. You can ignore this notification by selecting the `X` (**Clear Notification**).
 
 ## Explore IntelliSense
 
-In your new `helloworld.cpp` file, hover over `vector` or `string` to see type information. After the declaration of the `msg` variable, start typing `msg.` as you would when calling a member function. You should immediately see a completion list that shows all the member functions, and a window that shows the type information for the `msg` object:
+[IntelliSense](/editor/intellisense) is a tool to help you code faster and more efficently by adding code editing features such as code completion, parameter info, quick info, and member lists.
```
Comment (gregvanl) [inline docs/cpp/config-mingw.md:132]: /docs/editor/intellisense.md

**Comment by gregvanl** [comment]
@AlexandraKemperMS How does your update align with https://github.com/microsoft/vscode-docs/pull/6276? 
That community PR to update the installation to UCRT was approved but delayed for further engineering team testing. 
Should PR #6276 be closed in favor of this refresh?

File: docs/cpp/config-mingw.md [hunk 4]
```diff
@@ -3,58 +3,76 @@ Order: 3
...
+You can download the latest installer from the MSYS2 page or use this [**direct link to the installer**](https://github.com/msys2/msys2-installer/releases/download/2023-05-26/msys2-x86_64-20230526.exe).
 
-1. Install the Mingw-w64 toolchain (`pacman -S --needed base-devel mingw-w64-x86_64-toolchain`). Run the `pacman` command in a MSYS2 terminal. Accept the default to install all the members in the `toolchain` group.
+1. Run the installer and follow the steps of the installation wizard. Note that MSYS2 requires 64 bit Windows 8.1 or newer.
 
-1. Add the path to your Mingw-w64 `bin` folder to the Windows `PATH` environment variable by using the following steps:
-   1. In the Windows search bar, type 'settings' to open your Windows Settings.
+1. In the wizard, choose your desired Installation Folder. Record this directory for later. In most cases, the recommended directory is acceptable. The same applies when you get to setting the start menu shortcuts step. When complete, ensure the "Run MSYS2 now" box is checked and click "Finish". This will open a MSYS2 terminal window for you.
```
Comment (gregvanl) [inline docs/cpp/config-mingw.md:35]: Prefer using bold for UI strings so "ensure that **Run MSYS2 now** box is checked and select **Finish**.
Prefer "select" over "click" as some readers might not be using a mouse.

File: docs/cpp/config-mingw.md [hunk 5]
```diff
@@ -3,58 +3,76 @@ Order: 3
...
+
+1. In this terminal, install the MinGW-w64 toolchain by running the following command.
+```MSYS2
+pacman -S --needed base-devel mingw-w64-x86_64-toolchain
+```
+
+
+1.  Accept the default number of packages in the `toolchain` troup by pressing the `Enter` key.
```
Comment (gregvanl) [inline docs/cpp/config-mingw.md:43]: You can format raw keystrokes with `kbstyle()` so `kbstyle(Enter)`

File: docs/cpp/config-mingw.md [hunk 6]
```diff
@@ -105,23 +123,26 @@ Now press `kb(workbench.action.files.save)` to save the file. Notice how the fil
...
+
+To see IntelliSense in action,
+hover over `vector` or `string` to see their type information. If you type  `msg.` in line 10, you can see a completion list of recommended member functions to call, all generated by IntelliSense:
 
 ![Statement completion IntelliSense](images/wsl/msg-intellisense.png)
 
-You can press the `kbstyle(Tab)` key to insert the selected member; then, when you add the opening parenthesis, you will see information about any arguments that the function requires.
+You can press the `kbstyle(Tab)` key to insert a selected member. If you then add open parenthesis, IntelliSense will show information on which arguments are required.
+
+If IntelliSense is not already configured, open the command palette (`kb(workbench.action.showCommands)`) and enter **Select IntelliSense Configuration**. From the dropdown of compilers, select `Use gcc.exe` to configure.
```
Comment (gregvanl) [inline docs/cpp/config-mingw.md:141]: prefer "Command Palette" in upper case as it is VS Code "thing"

File: docs/cpp/config-mingw.md [hunk 7]
```diff
@@ -176,15 +201,15 @@ Your new `tasks.json` file should look similar to the JSON below:
...
 
 The `label` value is what you will see in the tasks list; you can name this whatever you like.
 
 The `detail` value is what you will as the description of the task in the tasks list. It's highly recommended to rename this value to differentiate it from similar tasks.
 
-From now on, the play button will read from `tasks.json` to figure out how to build and run your program. You can define multiple build tasks in `tasks.json`, and whichever task is marked as the default will be used by the play button. In case you need to change the default compiler, you can run **Tasks: Configure default build task**. Alternatively you can modify the `tasks.json` file and remove the default by replacing this segment:
+From now on, the play button will read from `tasks.json` to figure out how to build and run your program. You can define multiple build tasks in `tasks.json`, and whichever task is marked as the default will be used by the play button. In case you need to change the default compiler, you can run **Tasks: Configure Default Build Task** in the command palette. Alternatively you can modify the `tasks.json` file and remove the default by replacing this segment:
```
Comment (gregvanl) [inline docs/cpp/config-mingw.md:212]: "Command Palette"

File: docs/cpp/config-mingw.md (hunk 1 above)
Comment (AlexandraKemperMS) [reply docs/cpp/config-mingw.md]: Fixed.

File: docs/cpp/config-mingw.md (hunk 2 above)
Comment (AlexandraKemperMS) [reply docs/cpp/config-mingw.md]: Good to know - fixed throughout.

File: docs/cpp/config-mingw.md [hunk 8]
```diff
@@ -26,21 +26,21 @@ To successfully complete this tutorial, you must do the following steps:
...
 
 1. In this terminal, install the MinGW-w64 toolchain by running the following command.
 ```MSYS2
-pacman -S --needed base-devel mingw-w64-x86_64-toolchain
+pacman -S --needed base-devel mingw-w64-ucrt-x86_64-toolchain
 ```
 
 
-1.  Accept the default number of packages in the `toolchain` troup by pressing the `Enter` key.
+1.  Accept the default number of packages in the `toolchain` troup by pressing kbstyle(Enter).
```
Comment (gregvanl) [inline docs/cpp/config-mingw.md:43]: Nit: need to use backticks around kbstyle so `kbstyle(Enter)`

**Comment by AlexandraKemperMS** [comment]
> @AlexandraKemperMS How does your update align with #6276? That community PR to update the installation to UCRT was approved but delayed for further engineering team testing. Should PR #6276 be closed in favor of this refresh?

Good point, we discussed with the engineering team and it should be fine to add, so I included it in this PR. UCRT is only shipped with Windows 10 or later, so I added an additional question to the troubleshooting section in case users run into issues. Also engineering team might take a look at these doc changes to verify, I shared the link with them.

**Comment by gregvanl** [comment]
@AlexandraKemperMS Let me know when you'd like this merged after the engineering team has a look.

**Review by michelleangela — APPROVED** [review]

**Review by michelleangela — COMMENTED** [review]
⚠️    Do no merge yet until the extension is updated to look for the new UCRT path.

File: docs/cpp/config-mingw.md [hunk 9]
```diff
@@ -3,58 +3,78 @@ Order: 3
...
-1. Add the path to your Mingw-w64 `bin` folder to the Windows `PATH` environment variable by using the following steps:
-   1. In the Windows search bar, type 'settings' to open your Windows Settings.
+1. Run the installer and follow the steps of the installation wizard. Note that MSYS2 requires 64 bit Windows 8.1 or newer.
+
+1. In the wizard, choose your desired Installation Folder. Record this directory for later. In most cases, the recommended directory is acceptable. The same applies when you get to setting the start menu shortcuts step. When complete, ensure the **Run MSYS2 now** box is checked and select **Finish**. This will open a MSYS2 terminal window for you.
+
+1. In this terminal, install the MinGW-w64 toolchain by running the following command:
+   
+    ```MSYS2
+    pacman -S --needed base-devel mingw-w64-ucrt-x86_64-toolchain
```
Comment (bobbrow) [inline docs/cpp/config-mingw.md:41]: There should be a Note or something here about the availability of the ucrt requiring Windows 10.

**Comment by AlexandraKemperMS** [comment]
@bobbrow @michelleangela we can wait on adding UCRT until the add the support in the extension, for now we can just release the freshness pass on these docs and add UCRT at a later time.

File: docs/cpp/config-mingw.md (hunk 9 above)
Comment (AlexandraKemperMS) [reply docs/cpp/config-mingw.md]: This will be added back in when UCRT is added.

**Review by gregvanl — APPROVED** [review]

**Pull request merged**